pkg/rehearse/jobs_test.go

```diff
@@ -153,35 +155,40 @@ func makeTestingPresubmit(name, context string, ciopArgs []string, branch string
 
 func TestMakeRehearsalPresubmit(t *testing.T) {
 	testPrNumber := 123
-	testCases := []struct {
-		description string
-		source      *prowconfig.Presubmit
-		expected    *prowconfig.Presubmit
-	}{{
-		description: "sanity test",
-		source: makeTestingPresubmit(
-			"pull-ci-openshift-ci-operator-master-build",
-			"ci/prow/build",
-			[]string{"arg", "arg"},
-			"master"),
-		expected: makeTestingPresubmit(
-			"rehearse-123-pull-ci-openshift-ci-operator-master-build",
-			"ci/rehearse/openshift/ci-operator/master/build",
-			append([]string{"arg", "arg", "--git-ref=openshift/ci-operator@master"}),
-			"master"),
-	},
-	}
-	for _, tc := range testCases {
-		t.Run(tc.description, func(t *testing.T) {
-			rehearsal, err := makeRehearsalPresubmit(tc.source, "openshift/ci-operator", testPrNumber)
-			if err != nil {
-				t.Errorf("Unexpected error in makeRehearsalPresubmit: %v", err)
-			}
-			if !equality.Semantic.DeepEqual(tc.expected, rehearsal) {
-				t.Errorf("Expected rehearsal Presubmit differs:\n%s", diff.ObjectDiff(tc.expected, rehearsal))
-			}
-		})
+	testRepo := "org/repo"
+	sourcePresubmit := &prowconfig.Presubmit{
+		JobBase: prowconfig.JobBase{
+			Agent: "kubernetes",
+			Name:  "pull-ci-org-repo-branch-test",
+			Spec: &v1.PodSpec{
+				Containers: []v1.Container{{
+					Command: []string{"ci-operator"},
+					Args:    []string{"arg1", "arg2"},
+				}},
+			},
+		},
+		RerunCommand: "/test test",
+		Context:      "ci/prow/test",
+		Brancher:     prowconfig.Brancher{Branches: []string{"^branch$"}},
 	}
+	expectedPresubmit := &prowconfig.Presubmit{}
+	deepcopy.Copy(expectedPresubmit, sourcePresubmit)
+
+	expectedPresubmit.Name = "rehearse-123-pull-ci-org-repo-branch-test"
+	expectedPresubmit.Labels = map[string]string{rehearseLabel: "123"}
+	expectedPresubmit.Spec.Containers[0].Args = []string{"arg1", "arg2", "--git-ref=org/repo@branch"}
+	expectedPresubmit.RerunCommand = "/test pj-rehearse"
```

Comment:
Prow enforces that the compiled trigger regex matches the rerun command. This may just be on config load, but test this e2e before we merge to make sure it won't break anything, or set the trigger appropriately.

Reply:
I was aware of the check, and I tested it:

https://github.com/openshift/release/pull/2596#issuecomment-461896772